### PR TITLE
Add getEntity api

### PR DESCRIPTION
**Pull request by mcmuglim**
Provide getEntity() and getEntitlementInfo() for Context
Provide getChildren() for Entity
Provide getNextPage() for EntityList

R: @adzellman , @mttmllns , @HenryPing

### CODE REVIEW

File: www/CQMContext.js [hunk 1]
```diff
@@ -85,6 +121,52 @@ Entity.prototype._getImage = function(width, height, endpoint, successCallback,
...
+		};
+
+	var fail = errorCallback && function(code) {
+    	var ce = new CQMContextError(code);
+        errorCallback(ce);
+    };
+    exec(bind(this, success), fail, "CQMContext", "getNextPage", [this._parentId, this._parentVersion, this._parentVisibilityHash]);
+};
+
+EntityList.prototype._processRawEntities = function(rawEntities) {
```
Comment (mttmllns) [inline www/CQMContext.js:163]: What's this doing?  A doc would probably help.

File: www/CQMContext.js [hunk 2]
```diff
@@ -85,6 +121,52 @@ Entity.prototype._getImage = function(width, height, endpoint, successCallback,
 };
 
 /**
+ * This represents the an entity list.
```
Comment (mttmllns) [inline www/CQMContext.js]: "This represents ~~the~~ an entity list."

File: www/CQMContext.js [hunk 3]
```diff
@@ -85,6 +121,55 @@ Entity.prototype._getImage = function(width, height, endpoint, successCallback,
...
+	var success = successCallback && function(entityList) {
+			this.entities = this._processRawEntities(entityList.entities);
+			this.hasNextPage = entityList.hasNextPage;
+			successCallback(entityList);
+		};
+
+	var fail = errorCallback && function(code) {
+    	var ce = new CQMContextError(code);
+        errorCallback(ce);
+    };
```
Comment (mttmllns) [inline www/CQMContext.js:156]: There are some inconsistent spaces/tabs in this file.

File: www/CQMContext.js [hunk 4]
```diff
@@ -122,5 +207,87 @@ CQMContext.prototype.getInfo = function(successCallback, errorCallback) {
...
+	var entityNameRegex = /^[a-zA-Z0-9]+(?:[a-zA-Z0-9|_|-|.]*[a-zA-Z0-9]|[a-zA-Z0-9]*)$/;
+
+	if (entityName.length < 1 ||
+		entityName.length > 64 ||
+		entityNameRegex.test(entityName) == false) {
+		throw TypeError('Names must be limited to 64 characters. The value must start and end with a letter or number and can also contain dots, dashes, and underscores');
+	}
+
+
+	// Names must be limited to 64 characters. The value must start and end with a letter or number and can also contain dots, dashes, and underscore
```
Comment (mttmllns) [inline www/CQMContext.js]: Duplicate.

File: www/CQMContext.js (hunk 2 above)
Comment (mcmuglim) [inline www/CQMContext.js]: Fixed

File: www/CQMContext.js (hunk 4 above)
Comment (mcmuglim) [inline www/CQMContext.js]: Removed

File: www/CQMContext.js (hunk 3 above)
Comment (mcmuglim) [inline www/CQMContext.js:156]: Ah... yes! 4 spaces vs tabs everywhere. I didn't know which is the better practice. I went and changed everything to tabs.

Comment (adzellman) [inline www/CQMContext.js:156]: I think 4-spaces is the practice in most of the cordova plugins. For example, see https://github.com/apache/cordova-plugin-device/blob/master/www/device.js

Comment (mcmuglim) [inline www/CQMContext.js:156]: Alright. 4 spaces it is

**Comment by adzellman** [comment]
@mcmuglim Please confirm that this would qualify as a MINOR update according to semver: http://semver.org/

**Comment by mcmuglim** [comment]
@adzellman Yes, this is considered a minor update. New methods were added without changing existing ones.

**Pull request merged**